File: MeetingMinutesFrontend/PFPedro/src/app/services/email.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Temas } from '../clases/temas';

@Injectable({
  providedIn: 'root'
})
export class EmailService {

  private baseUrl = 'https://mm-project.herokuapp.com/email/';

  constructor(private http: HttpClient) { }

  /** Método para enviar la reunión a los participantes */

  enviarAgenda(receptores: any[], fechaReunion: string, temas: Temas[]): Observable<any> {
    return this.http.post(`${this.baseUrl}` + 'enviar/' + receptores + '/' + fechaReunion, temas);
  }

  /** Método para enviar el Acta a los asistenes */

  enviarActa(receptores: any[], fechaActa: string, temas: any[], tareas: any[], conclusion: string): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'enviaracta/' + fechaActa + '/' + receptores + '/' + temas + '/' + tareas + '/' + conclusion);
  }

  /** Método para enviar las tareas a los responsables */

  enviarTarea(codtarea: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'enviartarea/' + codtarea);
  }

  /** Método para enviar el comentario del Contáctanos a mi gmail */

  enviarComentario(nombre: string, correo: string, comentario: string): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'enviarcomentario/' + nombre + '/' + correo + '/' + comentario);
  }

}

File: MeetingMinutesBackend/MeetingMinutes/src/main/java/com/pedro/service/EmailService.java
/**
 * 
 */
package com.pedro.service;

import com.pedro.modelo.Temas;

/**
 * The Interface EmailService.
 *
 * @author Westermeyer
 */
public interface EmailService {
	
	/**
	 * Enviar agenda.
	 *
	 * @param receptores the receptores
	 * @param fechaReunion the fecha reunion
	 * @param tema the tema
	 */
	public void enviarAgenda(int[] receptores, String fechaReunion, Temas[] tema);
	
	/**
	 * Enviar acta.
	 *
	 * @param receptores the receptores
	 * @param fechaActa the fecha acta
	 * @param codtema the codtema
	 * @param codtarea the codtarea
	 * @param conclusion the conclusion
	 */
	public void enviarActa(int[] receptores, String fechaActa, int[] codtema, int[] codtarea, String conclusion);
	
	/**
	 * Enviar tarea.
	 *
	 * @param codtarea the codtarea
	 */
	public void enviarTarea(int codtarea);
	
	/**
	 * Enviar comentario.
	 *
	 * @param nombre the nombre
	 * @param correo the correo
	 * @param comentario the comentario
	 */
	public void enviarComentario(String nombre, String correo, String comentario);
	
}
